tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,697 @@
+- [Testing Strategy](#testing-strategy)
+- [Documentation Updates](#documentation-updates)
+- [Metrics/Telemetry](#metricstelemetry)
+
+---
+
+## Definitions
+
+- **Stage 6**: The standardized SDK preparation stage after code/tests/samples validation and before creating the release preparation PR.
+- **Metadata and Documentation Update**: Consolidated set of actions to ensure non-source assets (README, changelog, CI config, version numbers, additional descriptors) are accurate and consistent.
+- **Tool**: A discrete CLI (and MCP agent-exposed) command implementing one cohesive post-generation preparation task.
+- **Inline Script Command**: A single shell command provided directly in config that overrides a script path.
+
+---
+
+## Background / Problem Statement
+
+### Current State
+
+After SDK code generation, build, tests, and sample validation, engineers must perform repetitive cross-language housekeeping: updating changelogs, synchronizing README and language-specific files, adjusting package versions, exporting API surfaces (.NET), and inserting or validating CI configuration. Today these tasks are:
+
+- Partially automated in ad-hoc scripts per language / repo.
+- Inconsistent between management-plane and data-plane SDKs (especially version bump timing and changelog handling).
+- Error-prone when performed manually (missed version increments, stale README examples, inconsistent metadata presented across different files, like `pom.xml`, `_metadata.json`).
+- A barrier for integrating an AI agent workflow because required parameters and step order are not encapsulated as composable tools.
+
+### Why This Matters
+
+Without a unified tool layer:
+
+- Contributors spend time re-learning differing per-language conventions.
+- CI or release pipelines fail late due to missing metadata or mismatched versions.
+- Agents cannot reliably assist because state transitions ("now bump version", "now update changelog") aren't formalized.
+- Management-plane vs data-plane discrepancies introduce confusion (e.g., when to bump versions or generate changelog entries).
+
+Formalizing Stage 6 as four cohesive tools improves reliability, supports automation-first workflows, and prepares for future LLM-assisted changelog generation.
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+- [ ] Provide four composable, idempotent tools exposed via CLI and MCP agent: CI Update, Changelog Update, Version Update, Metadata Update.
+- [ ] Normalize execution order and expected outputs across all supported languages.
+- [ ] Encode management-plane vs data-plane behavioral differences explicitly (especially version & changelog logic).
+- [ ] Emit structured JSON results (status + message + next steps hints) enabling orchestration by AI agents and pipelines.
+- [ ] Support configuration-driven per-language scripting (path or inline) without recompiling the tool.
+
+### Exceptions and Limitations
+
+#### Exception 1: Data-Plane Deferred Version Bump
```

Comment:
I think our goal is to make data plane and mgmt. plane work to be consistent.
So curious is there any specific reason that those work could not be done earlier in the process?

cc @lmazuel

Reply:
@lirenhe:  Data plane packages follow a very different development lifecycle than management packages.   They generally do not release after a single pull request, and changes accumulate in `main` over a period of time.    The release date and version are not decided upfront - they're decided on after architect review and feedback.     The choice of version is an outcome of architect discussion and is nuanced; not something that can be calculated algorithmically.

There's no alignment in the process and approach between management and data plane packages.   Thus, we do not have the opportunity to force consistency in the tooling.